### ISSUE TITLE
provide repo_add_autopattern symbol with -DDEBIAN=1 -DENABLE_SUSEREPO

### ISSUE THREAD

**Issue by sunweaver** [openSUSE/libsolv#88]
The libsolv version in Debian will now switch to -DDEBIAN (was: -DSUSE). However, when doing so, the repo_add_autopattern symbols disappears.

In ext/CMakeLists.txt, the relating file repo_autopattern.(c,h) only get built/installed when -DSUSE is used at build time. Wouldn't it make more sense to provide that file if -DENABLE_SUSEREPO is set to 1 / is defined?

Patch will follow soon.

**Comment by mlschroe** [comment]
Not really. -DENABLE_SUSEREPO is for enabling the parser of the "susetags" format. The autopattern code is actually independent of the repository format, it's also used for repomd. As it's so limited to SUSE systems I tied it to -DSUSE, the more correct way would have been to support -DENABLE_AUTOPATTERN

**Comment by sunweaver** [comment]
----- Original message -----
> Not really. -DENABLE_SUSEREPO is for enabling the parser of the
> "susetags" format. The autopattern code is actually independent of the
> repository format, it's also used for repomd. As it's so limited to SUSE
> systems I tied it to -DSUSE, the more correct way would have been to
> support -DENABLE_AUTOPATTERN

Ok, I tied this to ENABLE_RPMMD then for now. Thanks.

**Comment by sunweaver** [comment]
http://anonscm.debian.org/cgit/collab-maint/libsolv.git/tree/debian/patches/1005_provide-repo-autopattern-for-non-SUSE-builds.patch
